### DISCUSSION TITLE
FetchContent calls in dependencies

### DISCUSSION THREAD

**Discussion by Leon0402** [cpm-cmake/CPM.cmake#323] [Q&A]
Hi, 

I do include some libraries that use FetchContent themself. I wonder if there is a way to use the cache for these as well?
It's probably rather difficult I could imagine, but maybe there is some way.

**Comment by TheLartians** [comment]
Hey sorry for the late answer, I hope this is still useful. 

I believe a solution would be to call [`FetchContent_Declare`](https://cmake.org/cmake/help/latest/module/FetchContent.html#command:fetchcontent_declare) using `SOURCE_DIR ${<library name>_SOURCE_DIR}` argument after adding the library using CPM with the [`DOWNLOAD_ONLY`](https://github.com/cpm-cmake/CPM.cmake#usage) option. This should overwrite any later calls for the same library, making `FetchContent_MakeAvailable` or `FetchContent_Populate` use the source provided by CPM.cmake.

**Comment by ClausKlein** [comment]
`set(FETCHCONTENT_SOURCE_DIR_<LIBRARY_NAME> ${<library name>_SOURCE_DIR})` should be enough.

**Reply by ClausKlein** [reply]
```cmake
CPMAddPackage(
  NAME Boost
  VERSION 1.81.0
  GITHUB_REPOSITORY "boostorg/boost"
  GIT_TAG "boost-1.81.0"
)
if(Boost_ADDED)
  set(FETCHCONTENT_SOURCE_DIR_BOOST ${Boost_SOURCE_DIR})
endif()
```
### but what, if an dependency need an other version?

### or expected the source archive contents tree, not the git repo with distributed include directories?
